### PR TITLE
feat: added onefetch

### LINKED ISSUE #1055: Requesting a new app: Onefetch missing from the `*fetch`  collection
## Requesting a new app ✨

If you are requesting a new app to be added to `deb-get` please complete the
following section, only one app per issue please.

 - [x] I have checked that this app does not already have a pending request and has not previously been declined.
 - App Name: Onefetch
 - Download URL: https://github.com/o2sh/onefetch
 - Website URL: https://github.com/o2sh/onefetch
 - App Summary: A command-line Git information tool written in Rust that displays project information and code statistics for a local Git repository directly to your terminal working entirely offline.

### PR DESCRIPTION
**Pull request by philclifford**
closes #1055

Co-authored-by: rasa  <rasa@users.noreply.github.com

### CODE REVIEW

File: 01-main/packages/onefetch [hunk 1]
```diff
@@ -0,0 +1,9 @@
+DEFVER=1
+get_github_releases "o2sh/onefetch" "latest"
+if [ "${ACTION}" != "prettylist" ]; then
+    URL=$(grep "browser_download_url.*\.deb\"" "${CACHE_FILE}" | head -n1 | cut -d'"' -f4)
```
Comment (rasa) [inline 01-main/packages/onefetch:4]: Per https://stackoverflow.com/a/17086464 we can extract the URL with a single command:
`URL=$(sed -E -n '/^.*browser_download_url.*"([^"].+\.deb)".*/{s//\1/p;q}' "${CACHE_FILE}")` 
Though it's admittedly more complex.
Also
`VERSION_PUBLISHED="$(echo "${URL}" | cut -d'/' -f8)"`
can be shortened to
`VERSION_PUBLISHED=$(cut -d'/' -f8 <<<"${URL}")`
Note the enclosing quotes are never needed when executing a command.

Comment (philclifford) [reply 01-main/packages/onefetch:4]: More complex, more painful to parse at 2am, harder to maintain 2 years later when upstream introduces RiscV and Arm debs or github api gets a subtle rev, and most importantly not helpful to the first time potential contributor just trying to do something entertaining as the their first foray into `bash` copy-pasta programming.

```
real	0m0.004s
user	0m0.004s
sys	    0m0.000s
```
is a significant improvement on
```
real	0m0.009s
user	0m0.007s
sys	    0m0.007s
```
but hardly time to make a cuppa so I'll stick with readability here - but thanks for the `sed` trick ! TIL

File: 01-main/packages/onefetch [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+    VERSION_PUBLISHED="$(echo "${URL}" | cut -d'/' -f8)"
```
Comment (philclifford) [inline 01-main/packages/onefetch:5]:
```suggestion
    VERSION_PUBLISHED=$(cut -d'/' -f8 <<<"${URL}")
```

File: 01-main/packages/onefetch (hunk 1 above)
Comment (philclifford) [reply 01-main/packages/onefetch:4]: Thanks @rasa . I very much appreciate and value your helpful review. :star:

Comment (rasa) [reply 01-main/packages/onefetch:4]: You might consider simplifying to
`URL=$(grep -m 1 "browser_download_url.*\.deb\"" "${CACHE_FILE}" | cut -d'"' -f4)`
as it's faster, removes complexity, and doesn't obfuscate. But no offense taken if you're not keen on this sort of optimization.

Comment (philclifford) [reply 01-main/packages/onefetch:4]: I love optimizations.  This is clear and I will adopt it.
Did you know you can make this kind of suggestion directly in a review comment (click a + on the line and then click the first button above the comment or type <ctrl-g> or directly enter with markdown

  ```suggestion
  URL=$(grep -m 1 "browser_download_url.*\.deb\"" "${CACHE_FILE}" | cut -d'"' -f4)
  ```
This allows the author/maintainer to easily adopt and commit your suggestion(s) and I think ensures you get a credit iirc.

**Review by flexiondotorg — APPROVED** [review]

**Pull request merged**